snap/quota/resources.go

```diff
@@ -189,15 +252,11 @@ func (qr *Resources) Validate() error {
 func (qr *Resources) ValidateChange(newLimits Resources) error {
 	// Check that the memory limit is not being decreased
 	if newLimits.Memory != nil {
-		if newLimits.Memory.Limit == 0 {
+		if qr.Memory != nil && newLimits.Memory.Limit == 0 {
 			return fmt.Errorf("cannot remove memory limit from quota group")
 		}
 
-		// make sure the memory limit is at least 640K, that is the minimum size
-		// we will require for a quota group. Newer systemd versions require up to
-		// 12kB per slice, so we need to ensure 'plenty' of space for this, and also
-		// 640kB seems sensible as a minimum.
-		if newLimits.Memory.Limit <= 640*quantity.SizeKiB {
+		if newLimits.Memory.Limit <= MemoryLimitMin {
 			return fmt.Errorf("memory limit %d is too small: size must be larger than 640KB", newLimits.Memory.Limit)
```

Comment:
Yea I was unsure about that because I can't do the .IECString() on the const values, and the alternative is to just let it print out the flat value '655360' which is not as nice as the previous text. Is there any cleaner way to do this without having to 'copy' the value into a new variable to do this?

Reply:
Actually you could probably tweak the quantity code to have `func (s Size) IECString() string` as it only makes sense to call it on values that exit, which should then work with consts too

Reply:
Thanks for your suggestion! I'll try it out!

Reply:
That worked!